packages/shared/src/lib/share.ts

```diff
@@ -30,3 +30,47 @@ export const getTelegramShareLink = (link: string, text: string): string =>
   `https://t.me/share/url?url=${encodeURIComponent(link)}&text=${text}`;
 export const getEmailShareLink = (link: string, subject: string): string =>
   `mailto:?subject=${subject}&body=${encodeURIComponent(link)}`;
+
+export const getShareLink = (
+  provider: ShareProvider,
+  link: string,
+  text?: string,
+): string => {
+  switch (provider) {
+    case ShareProvider.WhatsApp:
+      return getWhatsappShareLink(link);
+    case ShareProvider.Twitter:
+      return getTwitterShareLink(link, text ?? '');
+    case ShareProvider.Facebook:
+      return getFacebookShareLink(link);
+    case ShareProvider.Reddit:
+      return getRedditShareLink(link, text ?? '');
+    case ShareProvider.LinkedIn:
+      return getLinkedInShareLink(link);
+    case ShareProvider.Telegram:
+      return getTelegramShareLink(link, text ?? '');
+    case ShareProvider.Email:
+      return getEmailShareLink(link, text ?? '');
+    default:
+      return link;
+  }
+};
```

Comment:
Non blocking, but this looks like a great example of where unit tests would help just so we don't screw up the logic at any point now that we have it per provider.